can/interfaces/canalystii.py

```diff
@@ -69,132 +57,355 @@ class VCI_CAN_OBJ(Structure):
+        ('ArLost_ErrData', c_byte)
+    ]
+
+
+_in_use = {}
+
+
 class CANalystIIBus(BusABC):
     def __init__(
         self,
-        channel,
-        device=0,
-        bitrate=None,
-        Timing0=None,
-        Timing1=None,
+        channel: int,
+        device: int = 0,
+        bitrate: Optional[int] = None,
+        Timing0: Optional[int] = None,
+        Timing1: Optional[int] = None,
         can_filters=None,
         **kwargs,
     ):
         """
+        CANalystII Interface
+
+        :param channel: channel number to use
+        :type channel: int
+
+        :param device: device index. If there is more then one adapter the index number will be
+            incremented by one for each additional adapter.
+        :type device: int
+
+        :param bitrate: CAN network bandwidth (bits/s).
+        :type bitrate: int, optional
+
+        :param Timing0: customize the timing register if bitrate is not specified.
+        :type Timing0: int, optional
+
+        :param Timing1: customize the timing register if bitrate is not specified.
+        :type Timing1: int, optional
 
-        :param channel: channel number
-        :param device: device number
-        :param bitrate: CAN network bandwidth (bits/s)
-        :param Timing0: customize the timing register if bitrate is not specified
-        :param Timing1:
         :param can_filters: filters for packet
         """
-        super().__init__(channel=channel, can_filters=can_filters, **kwargs)
 
-        if isinstance(channel, (list, tuple)):
-            self.channels = channel
-        elif isinstance(channel, int):
-            self.channels = [channel]
-        else:
-            # Assume comma separated string of channels
-            self.channels = [int(ch.strip()) for ch in channel.split(",")]
+        if CANalystII is None:
+            raise RuntimeError('CANalystII library failed to load.')
 
-        self.device = device
+        super().__init__(channel=channel, can_filters=can_filters, **kwargs)
 
-        self.channel_info = "CANalyst-II: device {}, channels {}".format(
-            self.device, self.channels
+        self.channel = channel
+        self.device = device
+        self.channel_info = "CANalyst-II: device {}, channel {}".format(
+            self.device,
+            self.channel
         )
 
         if bitrate is not None:
-            try:
+            if bitrate in TIMING_DICT:
                 Timing0, Timing1 = TIMING_DICT[bitrate]
-            except KeyError:
-                raise ValueError("Bitrate is not supported")
+            else:
+                raise ValueError("Bitrate is not supported ({0})".format(bitrate))
 
-        if Timing0 is None or Timing1 is None:
+        if None in (Timing0, Timing1):
             raise ValueError("Timing registers are not set")
 
+        if device not in _in_use:
+            try:
+                CANalystII.VCI_UsbDeviceReset(VCI_USBCAN2, self.device, 0)
```

Comment:
What does this do? Is it necessary? Maybe write a comment why this was added.

Reply:
There are more then one variation of the Canalyst-II devices. Some of them being clones. Because the dll and the so file are not being included with this library that means any one of the different variations can be used. I have found that not all of them have the UsbDeviceReset function. But I have found that the ones that do seem to play a little nicer when the device gets reset just before setting it up.